### ISSUE TITLE
NPE in copy methods when using OptionalDouble

### ISSUE THREAD

**Issue by pabad** [immutables/immutables#284]
I'd like to congratule you @elucash both for this excellent project and the energy you put into makng it even more awesome!

Generating vaue objects to represent an API resource, I found that copy methods (withXX) generated for OptionalDouble fields attempt to unbox potentially null Double references.

The following example:
```java
public class Test {
	public static void main(final String[] args) {

		// NPE 1
		final ImmutableRepro r = ImmutableRepro.builder().build();
		r.withSome(OptionalDouble.of(1.0));

		// NPE 2
		final ImmutableRepro r2 = ImmutableRepro.builder().some(1.0).build();
		r2.withSome(OptionalDouble.empty());
	}

	@Immutable
	interface Repro {
		OptionalDouble some();
	}
}
```

Will generate a method:
```java
  public final ImmutableRepro withSome(OptionalDouble optional) {
    @Nullable Double value = optional.isPresent() ? optional.getAsDouble() : null;
    if (Double.doubleToLongBits(this.some) == Double.doubleToLongBits(value)) return this;
    return new ImmutableRepro(value);
  }
```

Which will trigger NPEs when updating an empty value or when updating a value with empty.
May I suggest that the code uses Objects.equals on the Double objects so that it handles correctly both the current value being null, and the optional being empty?

**Comment by elucash** [comment]
Thank you for the kind words!
Yep, the bug have to be fixed

**Comment by elucash** [comment]
Just released the fix in v2.1.15, please, try it

**Comment by pabad** [comment]
I can confirm the fix works. Thank you!
